apps/full-site-editing/full-site-editing-plugin/editor-domain-picker/src/domain-picker-fse/index.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import DomainPicker, { Props as DomainPickerProps } from '@automattic/domain-picker';
+import type { DomainSuggestions } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import 'a8c-fse-common-data-stores';
+import { useSite, useCurrentDomain } from '../hooks/use-current-domain';
+
+const FLOW_ID = 'gutenboarding';
+
+export type Props = Partial< DomainPickerProps >;
+
+const DomainPickerFSE: React.FunctionComponent< Props > = ( props ) => {
+	const [ domainSearch, setDomainSearch ] = React.useState( '' );
+
+	const site = useSite();
+	const currentDomain = useCurrentDomain();
+
+	const search = ( domainSearch.trim() || site?.name ) ?? '';
+
+	// TODO: This should go to the domain or launch store.
+	const [ domainName, setDomainName ] = React.useState( currentDomain );
+
+	const handleDomainSelect = ( domain?: DomainSuggestions.DomainSuggestion ) => {
+		// TODO: store whole domain suggestion object here because it has product_id and other useful info
```

Comment:
This comment was carried over from https://github.com/Automattic/wp-calypso/pull/43799 by @alshakero. 

@razvanpapadopol I think that's how we can distinguish what DomainStore stores, and what LaunchStore stores. DomainStore stores the full domain suggestion object, LaunchStore stores the current domain string.

Reply:
Don't we need to store the domain object (not string) because we need the `product_id` for the cart?

Reply:
Both are stored. Domain **object** in Domain store. Domain **string** in Launch store. 

This comes from the fact that Plan store has `currentPlan/setPlan`, but Domain store has no `currentDomain/setDomain`, which feels unbalanced. 

Plans Grid doesn't need outside intervention to set the current plan, while Domain Picker needs outside intervention to set the current domain.

Storing domain **string** in Launch store could make some sense as Launch store could retrieve the initial value from Site store.

Reply:
I'm not a huge fan on storing domain string in 2 places and I think it would make sense to have just a selector on the store to return the string instead of the whole object if needed. But not a blocker for now.